tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -224,69 +234,267 @@ func (p *PortTestSuite) TestClusterIP() {
 			Protocol:   corev1.ProtocolTCP,
 			Port:       int32(hostPort),
 			TargetPort: intstr.FromInt(defaultPort),
+			NodePort:   int32(nodePort),
 		},
 	}
-	clusterIPService := services.NewServiceTemplate(serviceName, p.namespace.Name, corev1.ServiceTypeClusterIP, ports, daemonsetTemplate.Spec.Template.Labels)
+	lbService := services.NewServiceTemplate(serviceName, p.namespace.Name, corev1.ServiceTypeLoadBalancer, ports, daemonsetTemplate.Spec.Selector.MatchLabels)
+	serviceResp, err := services.CreateService(steveClient, lbService)
+	require.NoError(p.T(), err)
+
+	err = services.VerifyService(steveClient, serviceResp)
+	require.NoError(p.T(), err)
+
+	p.validateLoadBalance(steveClient, nodePort, daemonsetName)
+}
+
+func (p *PortTestSuite) TestClusterIPScaleAndUpgrade() {
+	subSession := p.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(p.client, p.cluster.ID)
+	require.NoError(p.T(), err)
```

Comment:
can we move this to setup?

Reply:
we can't do that. For the upgrade test, we need to check the number of running pods, and to avoid conflicts with other pod tests, I'm using different namespaces for the upgrade tests